commit 708fb8c35eff3acacf2b7d197f9ef59fb9551e15
Author: Panu Matilainen <pmatilai@redhat.com>
Date:   Thu Jul 3 13:31:34 2014 +0300

    Fix db iterator pattern matching (RhBug:1115824)
    
    - Should've been in / Fixes a regression introduced in (pick your favorite)
      commit bc871e8fead5fd16f2c51a4a453bddf9994b80ba

diff --git a/lib/rpmdb.c b/lib/rpmdb.c
index be08193..b6d3247 100644
--- a/lib/rpmdb.c
+++ b/lib/rpmdb.c
@@ -1556,9 +1556,7 @@ top:
      * Skip this header if iterator selector (if any) doesn't match.
      */
     if (mireSkip(mi)) {
-	if (mi->mi_set)
-	    goto top;
-	return NULL;
+	goto top;
     }
     headerSetInstance(mi->mi_h, mi->mi_offset);
 
